plugins/modules/lxd_storage_volume_info.py

```diff
@@ -0,0 +1,461 @@
+        """Build URL with project parameter if specified."""
+        if self.project:
+            return f"{endpoint}?{urlencode({'project': self.project})}"
+        return endpoint
+
+    def _check_pool_exists(self) -> None:
+        """Verify that the storage pool exists."""
+        url = self._build_url(f"/1.0/storage-pools/{quote(self.pool, safe='')}")
+        resp_json = self.client.do("GET", url, ok_error_codes=[404])
+
+        if resp_json["type"] == "error":
+            if resp_json.get("error_code") == 404:
+                self.module.fail_json(msg=f'Storage pool "{self.pool}" not found')
+            else:
+                self.module.fail_json(
+                    msg=f'Failed to retrieve storage pool "{self.pool}": {resp_json.get("error", "Unknown error")}'
+                )
+
+    def _get_volume_list(self) -> list[str]:
+        """Get list of all volume URLs in the storage pool."""
+        url = self._build_url(f"/1.0/storage-pools/{quote(self.pool, safe='')}/volumes")
+        resp_json = self.client.do("GET", url, ok_error_codes=[])
+
+        if resp_json["type"] == "error":
+            self.module.fail_json(
+                msg=f'Failed to retrieve volumes from pool "{self.pool}": {resp_json.get("error", "Unknown error")}'
+            )
+
+        # The response contains a list of volume URLs like
+        # ['/1.0/storage-pools/default/volumes/custom/my-volume']
+        volume_urls = resp_json.get("metadata", [])
+        return volume_urls
+
+    def _parse_volume_url(self, volume_url: str) -> tuple[str, str]:
+        """Parse volume URL to extract type and name.
+
+        Expected URL format: /1.0/storage-pools/{pool}/volumes/{type}/{name}
+
+        The URL is split and empty parts (from leading/trailing slashes) are filtered out.
+        After normalization, the expected positions are:
+        - Index 0: "1.0"
+        - Index 1: "storage-pools"
+        - Index 2: pool name
+        - Index 3: "volumes"
+        - Index 4: volume type (e.g., "custom", "container", "virtual-machine", "image")
+        - Index 5: volume name
+
+        :param volume_url: The volume URL string from the LXD API
+        :type volume_url: str
+        :return: Tuple of (volume_type, volume_name)
+        :rtype: tuple[str, str]
+        :raises: Fails the module if the URL format is unexpected
+        """
+        # Split the URL and filter out empty parts from leading/trailing slashes
+        parts = [part for part in volume_url.split("/") if part]
+
+        # Validate we have the minimum expected number of parts
+        # We need at least: 1.0, storage-pools, {pool}, volumes, {type}, {name}
+        if len(parts) < 6:
+            self.module.fail_json(
+                msg=f'Unexpected volume URL format from LXD API: "{volume_url}". '
+                f"Expected at least 6 parts but got {len(parts)}. "
+                f"Expected format: /1.0/storage-pools/{{pool}}/volumes/{{type}}/{{name}}"
+            )
```

Comment:
Since this is used with API responses, and assuming the API will not change abruptly any time soon, it might make sense not to test and work under the assumption that the API will always return the information in the right format.